### ISSUE TITLE
Tikz Should Be Parsed As Raw By Default

### ISSUE THREAD

**Issue by schrieveslaach** [jgm/pandoc#3692]
The current dev version of pandoc parses the content of a tikz environment as text. For example, the content in the following example will be parsed as text.

```latex
Hello World

\begin{tikzpicture}
  \draw
    (3,-1) coordinate (a) node[right] {a}
    -- (0,0) coordinate (b) node[left] {b}
    -- (2,2) coordinate (c) node[above right] {c}
    pic["$\alpha$", draw=orange, <->, angle eccentricity=1.2, angle radius=1cm]
    {angle=a--b--c};
\end{tikzpicture}
```

Output of current pandoc dev version:

```
[Para [Str "Hello",Space,Str "World"]
,Para [Str "(3,-1)",Space,Str "coordinate",Space,Str "(a)",Space,Str "node[right]",Space,Span ("",[],[]) [Str "a"],SoftBreak,Str "\8211",Space,Str "(0,0)",Space,Str "coordinate",Space,Str "(b)",Space,Str "node[left]",Space,Span ("",[],[]) [Str "b"],SoftBreak,Str "\8211",Space,Str "(2,2)",Space,Str "coordinate",Space,Str "(c)",Space,Str "node[above",Space,Str "right]",Space,Span ("",[],[]) [Str "c"],SoftBreak,Str "pic[",Quoted DoubleQuote [Math InlineMath "\\alpha"],Str ",",Space,Str "draw=orange,",Space,Str "<->,",Space,Str "angle",Space,Str "eccentricity=1.2,",Space,Str "angle",Space,Str "radius=1cm]",SoftBreak,Span ("",[],[]) [Str "angle=a\8211b\8211c"],Str ";"]]
```

If you specify `-f latex+raw_tex`, it will be converted to:

```
[Para [Str "Hello",Space,Str "World"]
,RawBlock (Format "latex") "\\begin{tikzpicture}\\begin{tikzpicture}\n  \\draw\n    (3,-1) coordinate (a) node[right] {a}\n    -- (0,0) coordinate (b) node[left] {b}\n    -- (2,2) coordinate (c) node[above right] {c}\n    pic[\"$\\alpha$\", draw=orange, <->, angle eccentricity=1.2, angle radius=1cm]\n    {angle=a--b--c};\n\\end{tikzpicture}"]
```

However, when you specific this option and your latex code contains commands, which pandoc is not aware of, the parsing will fail. Often, these commands are not important to the pandoc conversion and should be ignored. Therefore, pandoc should parse tikz environments as `RawBlock` without a specific option.

**Comment by jgm** [comment]
>  your latex code contains commands, which pandoc is not aware of, the parsing will fail.

Can you give an example?  Maybe we should try to handle these cases better.

Although I'm tempted to do what you suggest here with `tikzpicture`, this would break an expectation some people might have:  that `-f latex-raw_tex` will produce an AST without raw tex.

**Comment by schrieveslaach** [comment]
> Can you give an example? Maybe we should try to handle these cases better.

I have a very long and complex latex document with many additional package and I assumed that pandoc gets confused by those unknown command. However, I took a closer look and I was wrong: `\ref` causes that parsing will fail (I think because of `unlessParseRaw` in the haskell code?). So, if the code above contains also `\ref` it fails.

In case of a user provided `-f latex-raw_tex`, pandoc code omit this `RawBlock`, right?

**Comment by jgm** [comment]
> So, if the code above contains also \ref, it fails.

Can you give exact input to reproduce?  I couldn't reproduce it by guessing.

> In case of a user provided -f latex-raw_tex, pandoc code omit this RawBlock, right?

Currently no -- it will try to parse the contents as text.  But we can easily change that so it will instead be omitted with a warning.

**Comment by jgm** [comment]
Oh I see the issue with your `ref` and `label` code.
When `unlessParseRaw` fails, the parser just raises an error.
We need something a bit fancier: check to see if the raw_tex extension is enabled; if it is, parse as raw, otherwise, do the other thing.  But I'll keep this issue closed; perhaps you should open a new issue or PR for this.

**Comment by jgm** [comment]
I'm going to fix this issue in the LaTeX reader.

**Comment by jgm** [comment]
Fixed by 8f718b08834e496e98790e1b5b8a3cb9e1b932a6
